packages/engine/README.md

```diff
@@ -229,6 +229,26 @@ In order to see more logging information while the simulation is running, you ca
 export RUST_LOG=debug
 ```
 
+If your simulation takes a lot of memory and is using JavaScript behaviors, the JavaScript runner might run out of memory.
+The first step you can take is passing a larger heap size to the runner:
+
+```shell
+cargo run --bin cli -- --project /path/to/my-hash-project single-run --num-steps <NUM-STEPS> --js-runner-max-heap-size <NEW-SIZE-IN-MB>
+```
+
+This will increase the heap size but you may still run into limitations.
+The next step is to recompile the underlying JavaScript engine and setting flags for it:
+
+```shell
+export V8_FROM_SOURCE = "1"
+export GN_ARGS = "v8_enable_pointer_compression=false v8_enable_shared_ro_heap=true"
+```
+
+- `V8_FROM_SOURCE` will force the V8 engine to be compiled from source and not use a pre-compiled version.
+This will take a fairly long time, expect 15min minimum. This can be mitigated in the subsequent compilations by using [sccache](https://github.com/mozilla/sccache) or [ccache](https://ccache.dev/). Our build scripts will detect and use them. Set the `$SCCACHE` or `$CCACHE` environmental variable if it's not in your path.
+- `v8_enable_pointer_compression` is an optimization reducing RAM usage at the cost of max heap size.
+- `v8_enable_shared_ro_heap` is required to disable pointer compression and allows the read-only memory to be shared across V8 isolate.
```

Comment:
```suggestion
- `v8_enable_shared_ro_heap` is required to disable pointer compression and enable read-only memory sharing by V8 isolates. This means, that read-only memory may be shared across different workers for JavaScript.
```

Reply:
We are creating an `Isolate` at each worker. There is one JS Runner per worker, each JS Runner runs `run_experiment` (_runner/javascript/mod.rs_, not _experiment/controller/run.rs_). I only clarified based on your description.

Reply:
I was wondering if creating a new platform would prevent this sharing between `Isolate`. But I guess not.

Reply:
What do you mean by "platform"?

Reply:
```rust
let platform = v8::new_default_platform(0, false).make_shared();
v8::V8::initialize_platform(platform);
```

Reply:
And I guess the platform is not shared across runners?
If this is an optimization we currently can't use we shouldn't mention the flag in the readme.

Reply:
I'm not actually sure if the platform is shared or not. At first I thought it wasn't but there might be only one and we initialize it if it's not already done.

We have to mention the flags, not because of its own value but because you can't disable pointer compression without it. V8 will not compile.

Reply:
> At first I thought it wasn't but there might be only one and we initialize it if it's not already done.

Could you figure this out?

> We have to mention the flag, not because of its own value but because you can't disable pointer compression without it. V8 will not compile.

This should be mentioned IMO.

Reply:
> Could you figure this out?

I've looked into it before replying earlier, let's just say the documentation is pretty sparse.
It seems to handle allocation, clock, async task.
So not a supervisor to `Isolate` as I first thought, more something unrelated.

Reply:
Verified in source: 

https://github.com/v8/v8/blob/eb9a19a0a11d7767888d632897f348ff36873841/src/heap/read-only-heap.cc#L68-L110

To summarize: With pointer compression, there is one `ReadOnlyHeap` per `Isolate`. Without PC, there is only one `ReadOnlyHeap` shared between all `Isolates`. There have a global `Mutex` and are completely independent from `Platform`, even though, `Platform` manages jobs.

> Mentioned V8 [5e5b0f7](https://github.com/hashintel/hash/commit/5e5b0f702018c5782f9a1c0d584d33087e0975ef)

I would put it the other way around: First describe, what this flag does, then say, that this is required if pointer compression is disabled. No strong opinion on that, though.

Reply:
Swapped the order https://github.com/hashintel/hash/pull/459/commits/1a4b19f24e996d0a65bae3c38ea89dff1c315edd